File: 02_DevApplication_IOS_avecObjectif-C/04_firstResponder/04_firstResponder/ViewController.m
//
//  ViewController.m
//  04_firstResponder
//
//  Created by etudiant on 4/19/17.
//  Copyright © 2017 etudiant. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextField *txt1;
@property (weak, nonatomic) IBOutlet UITextField *txt2;

@end

@implementation ViewController
- (IBAction)btnCacherClavierTxt1:(id)sender {
    [self.txt1 resignFirstResponder];
}
- (IBAction)btnCacherClavierTxt2:(id)sender {
    [self.txt2 resignFirstResponder];
}
- (IBAction)btnFocusSurTxt:(id)sender {
    [self.txt2 becomeFirstResponder];
}
//donner la possibilité à l'utilisateur de faire
//disparaitre le clavier visuel
//en touchant la partie livre de l'écran

//pour cela, je dois overrider la méthodes
//touches Began et arrêter l'édition en cours
//(où qu'elle soit)
-(void) touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    [self.view endEditing:YES];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 03_Swift_fondamentaux/02_DevAppIos/01_Devinette/03_firstResponder/03_firstResponder/ViewController.swift
//
//  ViewController.swift
//  03_firstResponder
//
//  Created by etudiant on 11/05/2017.
//  Copyright © 2017 etudiant. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var txt2: UITextField!
    @IBOutlet weak var txt1: UITextField!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    @IBAction func btnFocusSurTxt2(_ sender: UIButton) {
        txt2.becomeFirstResponder()
    }
   
    @IBAction func btnCacherClaviertxt2(_ sender: Any) {
        
            txt2.resignFirstResponder()
    }
    //cacher le clavier visuel lorsque j'utilise 
    //la touche return du clavier
    @IBAction func btnCacherClavierTxt1(_ sender: Any) {
        txt1.resignFirstResponder()
    }
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        //solution 1
        //demander au controleur de terminer 
        //toutes les opérations d"édition
        
        view.endEditing(true)
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

